subworkflows/local/qiime2_preptax.nf

```diff
@@ -2,20 +2,81 @@
  * Training of a classifier with QIIME2
  */
 
+include { UNTAR                 } from '../../modules/nf-core/untar/main'
+include { GZIP_DECOMPRESS       } from '../../modules/local/gzip_decompress.nf'
 include { FORMAT_TAXONOMY_QIIME } from '../../modules/local/format_taxonomy_qiime'
 include { QIIME2_EXTRACT        } from '../../modules/local/qiime2_extract'
 include { QIIME2_TRAIN          } from '../../modules/local/qiime2_train'
 
 workflow QIIME2_PREPTAX {
     take:
     ch_qiime_ref_taxonomy //channel, list of files
+    val_qiime_ref_taxonomy //val
     FW_primer //val
     RV_primer //val
 
     main:
-    FORMAT_TAXONOMY_QIIME ( ch_qiime_ref_taxonomy )
+    ch_qiime2_preptax_versions = Channel.empty()
+
+    if (params.qiime_ref_tax_custom) {
+        // Handle case where we have been provided a pair of filepaths.
+        if ("${params.qiime_ref_tax_custom}".contains(",")) {
+            ch_qiime_ref_taxonomy.flatten()
+                .branch {
+                    compressed: it.isFile() && it.getName().endsWith(".gz")
+                    decompressed: it.isFile() && ( it.getName().endsWith(".fna") || it.getName().endsWith(".tax") )
+                    failed: true
+                }.set { ch_qiime_ref_tax_branched }
+            ch_qiime_ref_tax_branched.failed.subscribe { error "$it is neither a compressed or decompressed sequence or taxonomy file. Please review input." }
```

Comment:
```suggestion
            ch_qiime_ref_tax_branched.failed.subscribe { error "$it is neither a compressed (ends with `.gz`) or decompressed sequence (ends with `.fna`) or taxonomy file (ends with `.tax`). Please review input." }
```
I think thats more transparent otherwise someone might be surprised their "*.fasta" isnt a decrompressed fasta file ;)